cli/pkg/local/app.go

```diff
@@ -483,6 +497,15 @@ func ParseLogFormat(format string) (LogFormat, bool) {
 	}
 }
 
+func checkPort(port int) error {
+	conn, err := net.Listen("tcp", ":"+strconv.Itoa(port))
+	if err != nil {
+		return err
+	}
+	defer conn.Close()
+	return nil
+}
```

Comment:
I'm still worried about the `Listen` here – it will actually open the port if it's not blocked. Could we do a Dial instead? Or is there something else we could do? I'm worried about the implications of opening the port twice in rapid succession. Sometimes it feels like the OS needs a bit of time to clean up ports.